docs/best-practices/continuous-integration/codemagic.md

```diff
@@ -0,0 +1,19 @@
+# Codemagic Integration
+
+[Codemagic](https://codemagic.io/) is a cloud-based CI/CD tool for mobile applications that you can use for continuous delivery together with *fastlane*. Codemagic has *fastlane* pre-installed, so you can easily run your *fastlane* scripts as part of the Codemagic build process. 
+
+When using *fastlane* for beta distribution to third-party services, you will need to add the API keys / secrets to Codemagic as [environment variables](https://docs.codemagic.io/building/environment-variables/). 
+
+Here's an example of using *fastlane* to publish a successfully built Android app to Firebase App Distribution as part of the Codemagic workflow.
+
+First, create a [Firebase token](https://firebase.google.com/docs/cli#cli-ci-systems), [encrypt](https://docs.codemagic.io/building/encrypting/) it and save it into an environment variable with the name `FIREBASE_TOKEN`.
+
+Then add the script to install the Fastlane Firebase App Distribution plugin and call the lane to publish the app. The scripts are added to the `publish` section of [codemagic.yaml](https://docs.codemagic.io/getting-started/yaml/) or as a pre-publish script when configuring the builds in the UI.
+
+```
+gem install bundler
+sudo gem install fastlane-plugin-firebase_app_distribution --user-install 

```

Comment:
Firebase documentation indicates users should be using fastlane to install this plugin [as per their documentation](https://firebase.google.com/docs/app-distribution/ios/distribute-fastlane):

```
fastlane add_plugin firebase_app_distribution
```

In which case would translate to `bundle exec fastlane add_plugin firebase_app_distribution` when using bundler.

Could you update this part of the documentation, and test to see if users won't face issues in Codemagic? 🙏

Reply:
I agree with @rogerluan. As the URL path explains, this list of CI integration pages should cover "the best practice" rather than basic instructions, like distribution beta apps using fastlane. "environment variables" would be a good example but others don't sound like "Codemagic" specific stuff.

@helinanever Do you have any other knowledge that you'd like to share with others? Please briefly go through what other CI integration documents cover as references and add what you have more🙇 
https://github.com/fastlane/docs/tree/master/docs/best-practices/continuous-integration

Reply:
@rogerluan @ainame thanks for the feedback, I will revisit this PR with the suggestions in mind to provide a document that adds more value.

Reply:
Thank you @helinanever! Feel free to tag me once this PR's ready to be reviewed again 🤗